File: app/src/main/java/de/nicidienase/chaosflix/shared/network/RecordingService.java
package de.nicidienase.chaosflix.shared.network;

import java.util.List;

import de.nicidienase.chaosflix.shared.entities.recording.Conference;
import de.nicidienase.chaosflix.shared.entities.recording.ConferencesWrapper;
import de.nicidienase.chaosflix.shared.entities.recording.Event;
import de.nicidienase.chaosflix.shared.entities.recording.Recording;
import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * Created by felix on 17.03.17.
 */

public interface RecordingService {

	@GET("public/conferences")
	Observable<ConferencesWrapper> getConferences();

	@GET("public/conferences/{id}")
	Observable<Conference> getConference(@Path("id") long id);

	@GET("public/events")
	Observable<List<Event>> getAllEvents();

	@GET("public/events/{id}")
	Observable<Event> getEvent(@Path("id") long id);

	@GET("public/recordings/{id}")
	Observable<Recording> getRecording(@Path("id") long id);

}

File: common/src/main/java/de/nicidienase/chaosflix/common/network/RecordingService.kt
package de.nicidienase.chaosflix.common.network

import de.nicidienase.chaosflix.common.entities.recording.Conference
import de.nicidienase.chaosflix.common.entities.recording.ConferencesWrapper
import de.nicidienase.chaosflix.common.entities.recording.Event
import de.nicidienase.chaosflix.common.entities.recording.Recording
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

interface RecordingService {

    @GET("public/conferences")
    fun getConferencesWrapper(): Single<ConferencesWrapper>

    @GET("public/events")
    fun getAllEvents(): Single<List<Event>>

    @GET("public/conferences/{id}")
    fun getConference(@Path("id") id: Long): Single<Conference>

    @GET("public/conferences/{name}")
    fun getConferenceByname(@Path("name") name: String): Single<Conference>

    @GET("public/events/{id}")
    fun getEvent(@Path("id") id: Long): Single<Event>

    @GET("public/recordings/{id}")
    fun getRecording(@Path("id") id: Long): Single<Recording>

}
